### PR TITLE
HAWQ-876. Add the support for initFile option of gpdiff.pl in hawq go…

### PR DESCRIPTION
**Pull request by paul-guo-**
…ogletest framework.

	This patch also removes several annoying warnings.

### CODE REVIEW

**Comment by wengyanqing** [comment]
LGTM

**Comment by ztao1987** [comment]
+1

File: src/test/feature/lib/hawq_config.cpp [hunk 1]
```diff
@@ -151,7 +151,7 @@ bool HawqConfig::isMultinodeMode() {
       psql.getQueryResult("select hostname from gp_segment_configuration");
   std::vector<std::vector<string> > table = result.getRows();
   std::unordered_set<string> hostnameMap;
-  for (int i = 0; i < table.size(); i++) {
+  for (unsigned int i = 0; i < table.size(); i++) {
```
Comment (xunzhang) [inline src/test/feature/lib/hawq_config.cpp]: use `size_t` instead

File: src/test/feature/lib/sql_util.cpp [hunk 2]
```diff
@@ -108,8 +111,24 @@ void SQLUtility::execSQLFile(const string &sqlFile,
...
+    // double check to avoid empty fileBaseName
+    if (fp.fileBaseName.empty())
+      ASSERT_TRUE(false) << initFileAbsPath << " is invalid";
+  } else {
+    initFileAbsPath = "";
+  }
+
+  bool is_sql_ans_diff = conn->checkDiff(ansFileAbsPath, outFileAbsPath, true, initFileAbsPath);
+  EXPECT_FALSE(is_sql_ans_diff);
+  if (is_sql_ans_diff == false) {
```
Comment (xunzhang) [inline src/test/feature/lib/sql_util.cpp]: `if (!is_sql_ans_diff) {`

Comment (xunzhang) [inline src/test/feature/lib/sql_util.cpp]: why you need to check here? You have already check it above: `EXPECT_FALSE(is_sql_ans_diff);`, I think the if-else clause is redundant.

**Comment by paul-guo-** [comment]
Merged before xunzhang commented.

File: src/test/feature/lib/sql_util.cpp (hunk 2 above)
Comment (paul-guo-) [inline src/test/feature/lib/sql_util.cpp]: That logic is for cleanup.

Comment (xunzhang) [inline src/test/feature/lib/sql_util.cpp]: You can directly remove it without if-else.

Comment (paul-guo-) [inline src/test/feature/lib/sql_util.cpp]: I think we can remove the else part, right?

 } else {
    EXPECT_FALSE(true);
  }

**Pull request closed without merging**